ui/src/app/pages/workspace/workspace-edit.tsx

```diff
@@ -1036,34 +1068,80 @@ export const WorkspaceEdit = fp.flow(withCurrentWorkspace(), withCdrVersions(),
                                 style={{marginTop: '24px'}} largeHeader
                                 required={!this.isMode(WorkspaceEditMode.Duplicate)}>
           <FlexRow style={{alignItems: 'baseline'}}>
-            <TextInput type='text' style={styles.textInput} autoFocus placeholder='Workspace Name'
-              value = {name}
-              onChange={v => this.setState(fp.set(['workspace', 'name'], v))}/>
-            <TooltipTrigger
+            <FlexColumn>
+              <div style={styles.fieldHeader}>
+                Workspace name
+              </div>
+              <TextInput data-test-id='workspace-name'
+                         type='text'
+                         style={styles.textInput}
+                         autoFocus
+                         placeholder='Workspace Name'
+                         value = {name}
+                         onChange={v => this.setState(fp.set(['workspace', 'name'], v))}/>
+            </FlexColumn>
+            {this.enableAccessTierSelection() &&
+              <FlexColumn>
+                <div style={styles.fieldHeader}>
+                  Data access tier
+                  <TooltipTrigger content={toolTipText.tierSelect}>
+                    <InfoIcon style={styles.infoIcon}/>
+                  </TooltipTrigger>
+                </div>
+                <TooltipTrigger
+                  content='To use a different access tier, create a new workspace.'
+                  disabled={this.isMode(WorkspaceEditMode.Create)}>
+                  <div data-test-id='select-access-tier' style={{...styles.select, ...styles.accessTierSpacing}}>
+                    <select style={{...styles.selectInput, ...styles.accessTierSpacing}}
+                            value={accessTierShortName}
+                            onChange={(v: React.FormEvent<HTMLSelectElement>) => {
+                              const selectedTier = v.currentTarget.value;
+                              this.setState(fp.set(['workspace', 'accessTierShortName'], selectedTier));
```

Comment:
opt nit: I think since there are 3 related state updates here, my preference would be to combine them into a single `setState`, e.g. via `fp.flow`, so that the update is atomic. I looked through the docs a bit, and I don't think this really has much of an impact in terms of re-rendering risk, nor did I find clear guidance on whether this pattern is generally preferred, so consider this comment optional.

Reply:
ok - now uses fp.flow